build/webapp-optimize.js

```diff
@@ -123,36 +116,12 @@ HTMLOptimizer.prototype._proceedLocales = function() {
 
     // create JSON dicts for the current language for locales-obj/
     if (this.config.GAIA_CONCAT_LOCALES === '1') {
-      this.asts[mozL10n.language.code] = this.getAST();
-    }
-    processedLocales++;
-  }
-
-  for (let lang in this.asts)  {
-    // skip to the next language if the AST is null
-    if (!this.asts[lang]) {
-      continue;
-    }
-    if (!this.webapp.asts[lang]) {
-      this.webapp.asts[lang] = [];
-    }
-    let asts = this.asts[lang];
-    let webappAsts = this.webapp.asts[lang];
-    for (let i = 0; i < asts.length; i++) {
-      let index = -1;
-      let identifierToFind = asts[i].$i;
-      for (let j = 0; j < webappAsts.length; j++) {
-        if (webappAsts[j].$i === identifierToFind) {
-          index = j;
-          break;
-        }
-      }
-      if (index !== -1) {
-        webappAsts[index] = asts[i];
-      } else {
-        webappAsts.push(asts[i]);
+      var ast = this.getAST();
+      if (ast.length !== 0) {
```

Comment:
exactly. Without this check I had this file in collection: https://github.com/mozilla-b2g/gaia/blob/master/apps/collection/add_to_collection.html creating an AST because it has l10n.js defined, but it was empty, because there are no resources linked.